### PR TITLE
Add test for IsUserAllowedToMerge with action user

### PR DESCRIPTION
**Pull request by lunny**
Add a test for #31056

### CODE REVIEW

File: services/pull/merge_test.go [hunk 1]
```diff
@@ -65,3 +72,16 @@ func Test_expandDefaultMergeMessage(t *testing.T) {
 		})
 	}
 }
+
+func Test_IsUserAllowedToMerge(t *testing.T) {
```
Comment (delvh) [inline services/pull/merge_test.go:76]:
```suggestion
func Test_ActionsUserCannotMerge(t *testing.T) {
```

**Review by delvh — APPROVED** [review]

**Comment by yp05327** [comment]
image
The root reason is here.
We only check the permission, and ignore the doer when white list is disabled.
Action user has the permission, but should not be allowed to merge.
So if you change the permission in test, it will fail.

**Review by yp05327 — CHANGES_REQUESTED** [review]
the test is not correct

**Comment by yp05327** [comment]
more details:
image
if you print the permission and the user, you can see action user has the permission, so `IsUserMergeWhitelisted` will return true.
image

**Comment by yp05327** [comment]
the write permission comes from here:
image

**Comment by lunny** [comment]
Replaced by #31173

**Pull request closed without merging**